### PR TITLE
added registration endpoint, and confirm password

### PR DESCRIPTION
**Pull request by GennadiyShpak**

### CODE REVIEW

File: src/app/auth/pages/confirm-password/confirm-password.component.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  constructor(
+    private router: Router,
+    private cognitoService: CognitoService,
+    private fb: FormBuilder
+  ) {}
+
+  ngOnInit(): void {
+    this.confirmForm = this.fb.group({
+      code: new FormControl('', { nonNullable: true }),
+      email: new FormControl('', { nonNullable: true })
```
Comment (Ponomaleks) [inline src/app/auth/pages/confirm-password/confirm-password.component.ts]: Why not use NonNullableFormBuilder here?

File: src/app/auth/pages/confirm-password/confirm-password.component.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormBuilder, FormControl, FormGroup, NonNullableFormBuilder, ReactiveFormsModule } from '@angular/forms';
```
Comment (Ponomaleks) [inline src/app/auth/pages/confirm-password/confirm-password.component.ts:3]: You don't need both of FormBuilder and NonNullableFormBuilder

Comment (GennadiyShpak) [reply src/app/auth/pages/confirm-password/confirm-password.component.ts:3]: i late, i`m already fixed it

File: src/app/auth/pages/confirm-password/confirm-password.component.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  constructor(
+    private router: Router,
+    private cognitoService: CognitoService,
+    private fb: NonNullableFormBuilder
+  ) {}
+
+  ngOnInit(): void {
+    this.confirmForm = this.fb.group({
+      code: '',
+      email: ''
```
Comment (Ponomaleks) [inline src/app/auth/pages/confirm-password/confirm-password.component.ts:39]: inputs shoild have at least required validation

File: src/app/auth/pages/confirm-password/confirm-password.component.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      code: this.code.value,
+      email: this.email.value
+    };
+    this.cognitoService
+      .confirmAuth(confirmFormValue)
+      .then(() => {
+        this.router.navigateByUrl(`/auth/${APP_ROUTER_NAME.LogIn}`);
+      })
+      .catch(() => {
+        this.router.navigateByUrl('/not-found');
```
Comment (HellraiserCW) [inline src/app/auth/pages/confirm-password/confirm-password.component.ts:54]: this.router.navigateByUrl(`/${APP_ROUTER_NAME.NotFound}`);

File: src/app/auth/pages/confirm-password/confirm-password.component.ts (hunk 2 above)
Comment (GennadiyShpak) [reply src/app/auth/pages/confirm-password/confirm-password.component.ts]: you are right

File: src/app/auth/pages/confirm-password/confirm-password.component.ts (hunk 3 above)
Comment (GennadiyShpak) [reply src/app/auth/pages/confirm-password/confirm-password.component.ts:39]: fixed

File: src/app/auth/pages/confirm-password/confirm-password.component.ts (hunk 4 above)
Comment (GennadiyShpak) [reply src/app/auth/pages/confirm-password/confirm-password.component.ts:54]: fixed

**Review by HellraiserCW — APPROVED** [review]

**Review by Ponomaleks — APPROVED** [review]

**Pull request merged**